README.md

```diff
@@ -846,12 +922,50 @@ then it **must** have a dependency like:
 }
 ```
 
-And, _even if_ your `<archetype>/package.json` also includes the exact same
-dependency.
+However, if you want to use builder to _also_ manage application dependencies,
+then you can follow [the module pattern](#the-module-pattern) and provide an
+`<archetype>/require.js` file consisting of:
+
+```js
+module.exports = require;
+```
+
+The root project could then require code like:
+
+
+```js
+var modFromProd = require("<archetype>/require")("lib-name");               // Module
+var pathFromProd = require("<archetype>/require").resolve("lib-name");      // Module path
+var modFromDev = require("<archetype-dev>/require")("lib-name");            // Module
+var pathFromDev = require("<archetype-dev>/require").resolve("lib-name");   // Module path
+```
+
+Using the above pattern, `<archetype>` or `<archetype-dev>` dependencies would
+override `<root>/node_modules` dependencies reliably and irrespective of npm
+flattening.
 
```

Comment:
This is actually a really good solution to providing app-dependencies in the archetype, because the code _explicitly_ states that the dependency is required from the archetype.

However, I don't see a good way to get this working for ES6-modules, so providing ES6 application dependencies in the archetype should not be recommended.